src/compiler/types.ts

```diff
@@ -2250,6 +2251,7 @@ namespace ts {
     /* @internal */
     export interface TypeCheckerHost {
         getCompilerOptions(): CompilerOptions;
+        getJsxFactoryEntity(): EntityName;

```

Comment:
this feels odd. i would rather we parsed the jsxFactory string one more time in the checker instead.